src/libstd/sys/common/wtf8.rs

```diff
@@ -434,19 +434,17 @@ impl fmt::Debug for Wtf8 {
             match self.next_surrogate(pos) {
                 None => break,
                 Some((surrogate_pos, surrogate)) => {
-                    try!(formatter.write_str(unsafe {
-                        // the data in this slice is valid UTF-8, transmute to &str
-                        mem::transmute(&self.bytes[pos .. surrogate_pos])
-                    }));
+                    // the data in this slice is valid UTF-8, transmute to &str
+                    let s: &str = unsafe { str::from_utf8_unchecked(&self.bytes[pos..surrogate_pos]) };
+                    try!(s.fmt(formatter));
```

Comment:
This should avoid the use of `fmt` directly as it will propagate the formatting attributes (like width) for each call to `fmt`, which probably isn't what we want in this case.

Also, won't this mean that lots of quote characters leak into the output? I thought we put quotes around the debug representation of strings..